src/olympia/devhub/tests/test_views.py

```diff
@@ -2819,26 +2821,26 @@ class TestAddVersion(AddVersionTest):
     def test_unique_version_num(self):
         self.version.update(version='0.1')
         r = self.post(expected_status=400)
-        assert_json_error(r, None, 'Version 0.1 already exists')
+        assert_json_error(
+            r, None, 'Version 0.1 already exists, or was uploaded before.')
 
     def test_same_version_if_previous_is_rejected(self):
-        # We can have several times the same version number, if the previous
+        # We can't re-use the same version number, even if the previous
         # versions have been disabled/rejected.
-        self.version.update(version='0.1', approvalnotes='approval notes')
-        self.version.releasenotes = 'release notes'
-        self.version.save()
+        self.version.update(version='0.1')
         self.version.files.update(status=amo.STATUS_DISABLED)
-        self.post(expected_status=200)
-
-        self.version.reload()
-        version = Version.objects.latest()
-        ok_(version.pk != self.version.pk)
-        eq_(version.version, self.version.version)
+        r = self.post(expected_status=400)
+        assert_json_error(
+            r, None, 'Version 0.1 already exists, or was uploaded before.')
 
-        # We reuse the release and approval notes from the last rejected
-        # version with the same version number.
-        eq_(version.releasenotes, self.version.releasenotes)
-        eq_(version.approvalnotes, self.version.approvalnotes)
+    def test_same_version_if_previous_is_deleted(self):
+        # We can't re-use the same version number if the previous
+        # versions has been deleted either.
+        self.version.update(version='0.1')
+        self.version.delete()
+        r = self.post(expected_status=400, nomination_type=3)
```

Comment:
Would be awesome if you could reuse the constant that already exists in `olympia.constants.base` (STATUS_NOMINATED)`